src/App.tsx

```diff
@@ -1,44 +1,82 @@
-import React from 'react';
+import React, { useState } from 'react';
 import 'bulma/bulma.sass';
 import '@fortawesome/fontawesome-free/css/all.css';
 import './App.scss';
 
 import classNames from 'classnames';
-import { PostsList } from './components/PostsList';
-import { PostDetails } from './components/PostDetails';
-import { UserSelector } from './components/UserSelector';
-import { Loader } from './components/Loader';
+import { PostsList } from './components/PostsList/PostsList';
+import { UserSelector } from './components/UserSelector/UserSelector';
+import { useSelectedUser } from './components/Contexts/UserContext';
+import { PostDetailsSidebar } from
+  './components/PostDetailsSidebar/PostDetailsSidebar';
+import { PostType } from './types/Post';
+import { User } from './types/User';
 
 export const App: React.FC = () => {
+  const [selectedPost, setSelectedPost] = useState<PostType | null>(null);
+
+  const [isNewCommentActive, setIsNewCommentActive] = useState(false);
+  const [isDropDownActive, setIsDropDownActive] = useState(false);
+
+  const [isPostsLoadingError, setIsPostsLoadingError] = useState(false);
+
+  const { selectedUser, setSelectedUser } = useSelectedUser();
+
+  const changeSelectedUser = (newUser: User) => {
+    setSelectedPost(null);
+    setSelectedUser(null);
+    setSelectedUser(newUser);
```

Comment:
Why do you set it to `null`, it doesn't work as expected in this case and makes no sense